R/comparePlot.R

```diff
@@ -96,35 +96,156 @@ comparePlot <- function(compareDF,
                         title = NULL,
-                        symbolSize = c(10, 10, 10, 4),
-                        symbolShape = c(21, 21, 21, 20),
-                        symbolColor = c("black", "grey0", "grey1", "grey25"),
-                        symbolFill = c("darkgoldenrod1", "deepskyblue4", "red3", "grey25"),
-                        alpha = 0.5,
+                        symbolSize,
+                        symbolShape,
+                        symbolColor = c("darkgoldenrod1", "deepskyblue4", "red3", "grey25"),
+                        transparency = 0.5,
                         crosshair = "grey50",
                         referenceLine = "darkgoldenrod1",
                         refLineThickness = 1,
                         legendPosition = "right",
                         footnote,
                         footnoteSize = 3,
-                        footnoteColor = "black"
-) {
-    set.seed(1954)
+                        footnoteColor = "black") {
     assertthat::assert_that(sum(apply(compareDF, 2, FUN = is.numeric)) >= 2,
-                            msg = "Need at least two numeric columns in compareDF.")
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
-                            msg = "Plot type must be either canvasXpress or ggplot.")
-    assertthat::assert_that(length(symbolSize) == 4,
-                            length(symbolShape) == 4,
-                            length(symbolColor) == 4,
-                            length(symbolFill) == 4,
-                            msg = paste0("All specified symbol arguments must be of length 4, ",
-                                         "including symbolSize, symbolShape, symbolColor, and symbolFill."))
+                           msg = "Need at least two numeric columns in compareDF.")
+    plotType <- tolower(plotType)
+    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
+                                       msg = "Plot type must be either canvasXpress or ggplot.")
+    if (any(is.null(pThreshold),
+            !is.numeric(pThreshold),
+            length(pThreshold) != 1)) {
+        warning("pThreshold must be a singular value of class numeric. Assigning default value '0.01'.")
+        pThreshold <- 0.01
+    }
+
+    if (!is.null(title) &&
+        !all(is.character(title), length(title) == 1)) {
+        warning("title must be a singular value of class character. Assigning default value 'NULL'.")
+        title <- NULL
+    }
+
+    if (!is.null(xlab) &&
+        !all(is.character(xlab), length(xlab) == 1)) {
+        warning("xlab must be a singular value of class character. Assigning default value 'NULL'.")
+        xlab <- NULL
+    }
+
+    if (!is.null(ylab) &&
+        !all(is.character(ylab), length(ylab) == 1)) {
+        warning("ylab must be a singular value of class character. Assigning default value 'NULL'.")
+        ylab <- NULL
+    }
+
+    if (any(is.null(symbolColor),
```

Comment:
There is a validation function that I have created in ggplotMDS .rgba_conversion. Use to validate the input color format.

Reply:
done